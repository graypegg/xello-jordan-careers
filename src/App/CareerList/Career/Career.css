.Career__wrapper {
  padding: 25px 25px 30px;
  margin: 20px;
  background: rgb(247, 247, 247);
  color: #333;
  line-height: 1.5;
}

.Career__wrapper > h1 {
  margin: 0 0 15px;
  color: var(--primary-light);
  background: #2a3a2d;
  text-align: center;
  padding: 15px;
}

.Career__wrapper > p {
  padding: 0 20px;
}

.Career__image {
  height: 350px;
  background-size: cover;
  background-position: center center;
}