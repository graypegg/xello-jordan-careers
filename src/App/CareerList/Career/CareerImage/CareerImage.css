.CareerImage__image {
  position: relative;
  height: 350px;
  background-size: cover;
  background-position: center center;
}

.CareerImage__modeToggleButton,
.CareerImage__modeToggleButton--active {
  position: absolute;
  bottom: 10px;
  right: 10px;
  display: inline-block;
  padding: 10px 15px;
  color: #FFF;
  border-radius: 15px;
  cursor: pointer;
}

.CareerImage__modeToggleButton {
  background: var(--primary-dark);
}

.CareerImage__modeToggleButton--active {
  background: var(--dark);
}

.CareerImage__modeToggleButton--active:hover {
  background: var(--darker);
}

.CareerImage__modeToggleButton:hover {
  background: var(--primary-light);
}