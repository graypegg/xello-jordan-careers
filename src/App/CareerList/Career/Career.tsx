import { ICareer } from '../../../types'
import * as React from 'react'

import './Career.css'

interface ICareerProps {
  career: ICareer,
  showImage: boolean
}

function Career(props: ICareerProps): JSX.Element {
  return (
    <div className="Career__wrapper">
      {props.showImage ? (
        <div className="Career__image" style={ {backgroundImage: `url(${props.career.image})`} } />
      ) : undefined}
      <h1>{props.career.title}</h1>
      <p>{props.career.description}</p>
      <ul>
        {props.career.notes.map((note) => (
          <li key={note}>
            {note}
          </li>
        ))}
      </ul>
    </div>
  )
}

export default Career