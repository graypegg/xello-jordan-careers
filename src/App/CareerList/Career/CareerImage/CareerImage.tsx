import { ImageMode } from '../../../../consts'

import * as React from 'react'

import './CareerImage.css'

interface ICareerImageProps {
  image: string
}

interface ICareerImageState {
  mode: ImageMode
}

class CareerImage extends React.Component<ICareerImageProps, ICareerImageState> {
  constructor (props: ICareerImageProps) {
    super(props)

    this.state = {
      mode: ImageMode.Low
    }

    this.scaleImage = this.scaleImage.bind(this)
    this.toggleMode = this.toggleMode.bind(this)
  }

  public scaleImage (image: string) {
    return `${image}${this.state.mode === ImageMode.High ? '' : '?quality=70&maxWidth=500'}`
  }

  public toggleMode (): void {
    this.setState({
      mode: this.state.mode === ImageMode.High ? ImageMode.Low : ImageMode.High
    })
  }

  public render (): JSX.Element {
    return (
      <div className="CareerImage__wrapper">
        <div className="CareerImage__image" style={ { backgroundImage: `url(${this.scaleImage(this.props.image)})` } }>
          <div
            className={this.state.mode === ImageMode.High ? 'CareerImage__modeToggleButton--active' : 'CareerImage__modeToggleButton'}
            onClick={this.toggleMode}>
              {
                this.state.mode === ImageMode.High
                ? 'Show Low Res'
                : 'Show High Res'
              }
          </div>
        </div>
      </div>
    )
  }
}

export default CareerImage