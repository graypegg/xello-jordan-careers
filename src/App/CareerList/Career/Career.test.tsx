import { ICareer } from '../../../types'

import * as React from 'react'
import { shallow } from 'enzyme'
import Career from './Career'
import CareerImage from './CareerImage/CareerImage'

const career: ICareer = {
  title: 'Test',
  description: 'Test',
  notes: ['a', 'b'],
  image: 'TestImage',
  id: 1
}

const onSaveBookmarkMock = (_: ICareer) => { /* */ }

it('renders without crashing', () => {
  shallow(<Career career={career} showImage={false} onSaveBookmark={onSaveBookmarkMock} />)
})

describe('can render a simple career', () => {
  
  const component = shallow(<Career career={career} showImage={false} onSaveBookmark={onSaveBookmarkMock} />)

  it('rendered a title with bookmark button', () => {
    expect(component.find('h1').text()).toBe(career.title)
    expect(component.find('h1 > .Career__bookmarkButton').length).toBe(1)
  })

  it('rendered a description', () => {
    expect(component.find('p').text()).toBe(career.description)
  })

  it('rendered a notes list', () => {
    expect(component.children('ul').find('li').at(0).text()).toBe(career.notes[0])
    expect(component.children('ul').find('li').at(1).text()).toBe(career.notes[1])
  })

  it('does render an image when props.showImages == true', () => {
    const componentWithImage = shallow(<Career career={career} showImage={true} onSaveBookmark={onSaveBookmarkMock} />)
    expect(componentWithImage.find(CareerImage).length).toBe(1)
    expect(componentWithImage).toMatchSnapshot()

  })

  it('does not render an image when props.showImages == false', () => {
    expect(component.find(CareerImage).length).toBe(0)
    expect(component).toMatchSnapshot()
  })
})