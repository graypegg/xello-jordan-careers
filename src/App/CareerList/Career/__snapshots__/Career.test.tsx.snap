// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render a simple career does not render an image when props.showImages == false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Career
    career={
      Object {
        "description": "Test",
        "id": 1,
        "image": "TestImage",
        "notes": Array [
          "a",
          "b",
        ],
        "title": "Test",
      }
    }
    onDeleteBookmark={[Function]}
    onSaveBookmark={[Function]}
    showImage={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <h1>
          <div
            className="Career__titleContainer"
          >
            Test
          </div>
          <div
            className="Career__bookmarkButton"
            onClick={[Function]}
          >
            <img
              src={undefined}
              width="50"
            />
          </div>
        </h1>,
        <p>
          Test
        </p>,
        <ul>
          <li>
            a
          </li>
          <li>
            b
          </li>
        </ul>,
        <a
          className="Career__profileLink"
          href="https://student.xello.world/options/career/1"
          target="_blank"
        >
          https://student.xello.world/options/career/
          1
        </a>,
      ],
      "className": "Career__wrapper",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="Career__titleContainer"
            >
              Test
            </div>,
            <div
              className="Career__bookmarkButton"
              onClick={[Function]}
            >
              <img
                src={undefined}
                width="50"
              />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test",
              "className": "Career__titleContainer",
            },
            "ref": null,
            "rendered": "Test",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                src={undefined}
                width="50"
              />,
              "className": "Career__bookmarkButton",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "src": undefined,
                "width": "50",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test",
        },
        "ref": null,
        "rendered": "Test",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              a
            </li>,
            <li>
              b
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "a",
            "nodeType": "host",
            "props": Object {
              "children": "a",
            },
            "ref": null,
            "rendered": "a",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "b",
            "nodeType": "host",
            "props": Object {
              "children": "b",
            },
            "ref": null,
            "rendered": "b",
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "https://student.xello.world/options/career/",
            1,
          ],
          "className": "Career__profileLink",
          "href": "https://student.xello.world/options/career/1",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Array [
          "https://student.xello.world/options/career/",
          1,
        ],
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <h1>
            <div
              className="Career__titleContainer"
            >
              Test
            </div>
            <div
              className="Career__bookmarkButton"
              onClick={[Function]}
            >
              <img
                src={undefined}
                width="50"
              />
            </div>
          </h1>,
          <p>
            Test
          </p>,
          <ul>
            <li>
              a
            </li>
            <li>
              b
            </li>
          </ul>,
          <a
            className="Career__profileLink"
            href="https://student.xello.world/options/career/1"
            target="_blank"
          >
            https://student.xello.world/options/career/
            1
          </a>,
        ],
        "className": "Career__wrapper",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="Career__titleContainer"
              >
                Test
              </div>,
              <div
                className="Career__bookmarkButton"
                onClick={[Function]}
              >
                <img
                  src={undefined}
                  width="50"
                />
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test",
                "className": "Career__titleContainer",
              },
              "ref": null,
              "rendered": "Test",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  src={undefined}
                  width="50"
                />,
                "className": "Career__bookmarkButton",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "src": undefined,
                  "width": "50",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test",
          },
          "ref": null,
          "rendered": "Test",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                a
              </li>,
              <li>
                b
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "a",
              "nodeType": "host",
              "props": Object {
                "children": "a",
              },
              "ref": null,
              "rendered": "a",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "b",
              "nodeType": "host",
              "props": Object {
                "children": "b",
              },
              "ref": null,
              "rendered": "b",
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "https://student.xello.world/options/career/",
              1,
            ],
            "className": "Career__profileLink",
            "href": "https://student.xello.world/options/career/1",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Array [
            "https://student.xello.world/options/career/",
            1,
          ],
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`can render a simple career does render an image when props.showImages == true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Career
    career={
      Object {
        "description": "Test",
        "id": 1,
        "image": "TestImage",
        "notes": Array [
          "a",
          "b",
        ],
        "title": "Test",
      }
    }
    onDeleteBookmark={[Function]}
    onSaveBookmark={[Function]}
    showImage={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <CareerImage
          image="TestImage"
        />,
        <h1>
          <div
            className="Career__titleContainer"
          >
            Test
          </div>
          <div
            className="Career__bookmarkButton"
            onClick={[Function]}
          >
            <img
              src={undefined}
              width="50"
            />
          </div>
        </h1>,
        <p>
          Test
        </p>,
        <ul>
          <li>
            a
          </li>
          <li>
            b
          </li>
        </ul>,
        <a
          className="Career__profileLink"
          href="https://student.xello.world/options/career/1"
          target="_blank"
        >
          https://student.xello.world/options/career/
          1
        </a>,
      ],
      "className": "Career__wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "image": "TestImage",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="Career__titleContainer"
            >
              Test
            </div>,
            <div
              className="Career__bookmarkButton"
              onClick={[Function]}
            >
              <img
                src={undefined}
                width="50"
              />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test",
              "className": "Career__titleContainer",
            },
            "ref": null,
            "rendered": "Test",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                src={undefined}
                width="50"
              />,
              "className": "Career__bookmarkButton",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "src": undefined,
                "width": "50",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test",
        },
        "ref": null,
        "rendered": "Test",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              a
            </li>,
            <li>
              b
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "a",
            "nodeType": "host",
            "props": Object {
              "children": "a",
            },
            "ref": null,
            "rendered": "a",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "b",
            "nodeType": "host",
            "props": Object {
              "children": "b",
            },
            "ref": null,
            "rendered": "b",
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "https://student.xello.world/options/career/",
            1,
          ],
          "className": "Career__profileLink",
          "href": "https://student.xello.world/options/career/1",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Array [
          "https://student.xello.world/options/career/",
          1,
        ],
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <CareerImage
            image="TestImage"
          />,
          <h1>
            <div
              className="Career__titleContainer"
            >
              Test
            </div>
            <div
              className="Career__bookmarkButton"
              onClick={[Function]}
            >
              <img
                src={undefined}
                width="50"
              />
            </div>
          </h1>,
          <p>
            Test
          </p>,
          <ul>
            <li>
              a
            </li>
            <li>
              b
            </li>
          </ul>,
          <a
            className="Career__profileLink"
            href="https://student.xello.world/options/career/1"
            target="_blank"
          >
            https://student.xello.world/options/career/
            1
          </a>,
        ],
        "className": "Career__wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "image": "TestImage",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="Career__titleContainer"
              >
                Test
              </div>,
              <div
                className="Career__bookmarkButton"
                onClick={[Function]}
              >
                <img
                  src={undefined}
                  width="50"
                />
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test",
                "className": "Career__titleContainer",
              },
              "ref": null,
              "rendered": "Test",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  src={undefined}
                  width="50"
                />,
                "className": "Career__bookmarkButton",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "src": undefined,
                  "width": "50",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test",
          },
          "ref": null,
          "rendered": "Test",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                a
              </li>,
              <li>
                b
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "a",
              "nodeType": "host",
              "props": Object {
                "children": "a",
              },
              "ref": null,
              "rendered": "a",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "b",
              "nodeType": "host",
              "props": Object {
                "children": "b",
              },
              "ref": null,
              "rendered": "b",
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "https://student.xello.world/options/career/",
              1,
            ],
            "className": "Career__profileLink",
            "href": "https://student.xello.world/options/career/1",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Array [
            "https://student.xello.world/options/career/",
            1,
          ],
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
